core/src/main/java/com/twitter/elephantbird/mapreduce/input/MultiInputFormat.java

```
@@ -91,7 +95,17 @@ public static void setGenericConverterClassConf(Class<?> clazz, Configuration co
     }
     Class<?> recordClass = typeRef.getRawClass();
 
-    Format fileFormat = determineFileFormat(split, conf);
+    FileSplit fileSplit = (FileSplit) split;
+
+    final Format fileFormat;
+    try {
+      LOG.info("Determining format of file: " + fileSplit.getPath());
+      fileFormat = determineFileFormat(fileSplit, conf);
+    } catch (IOException e) {
+      throw new IOException("Could not determine format of file: " + fileSplit.getPath(), e);
```

The useful part is the path is added to the exception. It will be passed to the AM and will appear in the job diagnostic info even on the submitter (and webUI). But I would not distinguish, just have a catch-all Exception.